### ISSUE TITLE
Poor barrier generation with multiple command buffers

### ISSUE THREAD

**Issue by JMS55** [gfx-rs/wgpu#6674]
Given:
* CommandBuffer A: Use resource X as a render pass attachment
* CommandBuffer B: Use resource Y as a render pass attachment
* CommandBuffer C: Use resources X and Y in a bind group

Wgpu generates:
* CommandBuffer 0: Barrier to transition resource X to attachment
* CommandBuffer A: Use resource X as a render pass attachment
* CommandBuffer 1: Barrier to transition resource Y to attachment
* CommandBuffer B: Use resource Y as a render pass attachment
* CommandBuffer 2: Barrier to transition resource X and Y to shader access
* CommandBuffer C: Use resources X and Y in a bind group

Ideally it would generate:
* CommandBuffer 0: Barrier to transition resource X and Y to attachment
* CommandBuffer A: Use resource X as a render pass attachment
* CommandBuffer B: Use resource Y as a render pass attachment
* CommandBuffer 1: Barrier to transition resource X and Y to shader access
* CommandBuffer C: Use resources X and Y in a bind group

This way there is no barrier between A and B, and they will be able to overlap.

The problem scales worse with the number of resources. This is an example trace from bevy with ~7 textures. Notice the large gap between each pass.

![image](https://github.com/user-attachments/assets/2605db6c-142d-40db-b39b-9538aaa7567b)

**Comment by JMS55** [comment]
Fixing the barriers didn't actually prevent the gaps. It might be because wgpu puts each renderpass in a separate command buffer, but I'm not sure.

**Comment by jimblandy** [comment]
Yeah, this kind of observation is essential, as we consider our options here.
